website/content/en/docs/best-practices/multi-tenancy.md

```diff
@@ -0,0 +1,72 @@
+[Requests and limits][k8s-requests-limits]
+
+**IMPORTANT:** Because OLM-managed Operators typically don't run on Cluster infra nodes, they are consuming paid CPU cores from a user
+perspective, so they naturally want to at least limit the maximum amount of resources such non-workload components can take up
+For this reason, if you are integrating your project with [OLM][olm] you should **always ship your operator `Deployment` with resource requests**.
+
+The concepts above also apply to tenant workloads which are managed by your operator. Cluster administrators will typically set
+a `ResourceQuota` for each tenant's namespace / project as part of the onboarding. If your operator creates `Pods` inside
+tenant namespace which do not specify at least resource requests for CPU and RAM, those pods will fail to schedule and tenants
+will not be able to use your operator. Therefore your operator should **always apply at least resource requests for CPU and
+RAM** for `Pods` that it creates as part of reconcilication. Ideally your operator also applies limits to those `Pods`.
+Resource requests are also needed for horizontal pod autoscaling to work
+([docs](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-resource-metrics)).
+
+### If your operator creates `Ingress` objects
+
+* allow users to supply custom labels to `Ingress` to allow for router sharding
+
+This is a popular configuration option for dense multi-tenant clusters, where several [IngressController][k8s-ingress-controllers] configuration have
+been deployed. The goal is to split ingress traffic from e.g. production and development, ending up on different routers. These
+routers are `Pods` inside a `Deployment` that carry a specific label to receive traffic from certain ingress controllers. The
+label is applied at the `Ingress` level and there is no pre-defined convention here, so users set these custom labels in
+accordance to how they configured their `IngressController` instances.
+
+**NOTE** If you are developing your Operator to be used in K8S vendor like OpenShift or OKD you might want to use helper APIs to work 
+with `Ingress` such as, for example, [Route][ocp-route]. In this way, we would like to recommend check its [docs][ocp-ingress-doc]
+and this [blog][ocp-blog] for an end-to-end example.
+
+**Your operator should allow the user to specify custom labels for any `Ingress` that it manages**.
+
+### If your operator creates `NetworkPolicy` objects
+
+* apply fine-grained network policies to the extent that is required for your managed application to function properly
+* apply fine-grained network policies to enable your managed application internal components to communicate among each other
+* allow users to disable network policy management (your operator doesn't deploy [NetworkPolicy][k8s-network-policy] instances at all)
+* do not create "allow traffic from everywhere in the cluster" type policies
```

Comment:
```suggestion
If your operator creates `NetworkPolicy` objects 

* apply fine-grained network policies to the extent that is required for your managed application to function properly
* apply fine-grained network policies to enable your managed application internal components to communicate among each other
* allow users to disable network policy management (your operator doesn't deploy [NetworkPolicy][k8s-network-policy] instances at all)
* do not create "allow traffic from everywhere in the cluster" type policies
```

Again _WHAT_ should apply fine-grained network policies? The Operator?

Reply:
please let me know if that is better now.